### PR TITLE
chore: upgrade webpack@5 to fix CI builds

### LINKED ISSUE #6502: CI builds are failing due to ERR_OSSL_EVP_UNSUPPORTED.
**Describe the bug**
There is an issue with hashing algorithms in Webpack 4 and Node versions 17 and above, due to Node’s migration to OpenSSL3. It’s documented at https://github.com/webpack/webpack/issues/14532 — and all recent builds on this repo are failing due to the issue. 

Builds fail with: 

```
Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:67:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (/home/runner/work/netlify-cms/netlify-cms/node_modules/webpack/lib/util/createHash.js:135:53)
    at NormalModule._initBuildHash (/home/runner/work/netlify-cms/netlify-cms/node_modules/webpack/lib/NormalModule.js:417:16)
    at handleParseError (/home/runner/work/netlify-cms/netlify-cms/node_modules/webpack/lib/NormalModule.js:471:10)
    at /home/runner/work/netlify-cms/netlify-cms/node_modules/webpack/lib/NormalModule.js:503:5
    at /home/runner/work/netlify-cms/netlify-cms/node_modules/webpack/lib/NormalModule.js:358:12
    at /home/runner/work/netlify-cms/netlify-cms/node_modules/loader-runner/lib/LoaderRunner.js:373:3
    at iterateNormalLoaders (/home/runner/work/netlify-cms/netlify-cms/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
    at iterateNormalLoaders (/home/runner/work/netlify-cms/netlify-cms/node_modules/loader-runner/lib/LoaderRunner.js:221:10)
    at /home/runner/work/netlify-cms/netlify-cms/node_modules/loader-runner/lib/LoaderRunner.js:236:3
    at context.callback (/home/runner/work/netlify-cms/netlify-cms/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at /home/runner/work/netlify-cms/netlify-cms/node_modules/babel-loader/lib/index.js:59:71 {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}
``` 

Example pull requests which have failed with this error: 

- https://github.com/netlify/netlify-cms/pull/6498
- https://github.com/netlify/netlify-cms/pull/6496
- https://github.com/netlify/netlify-cms/pull/6492

**To Reproduce**
1. Make a change (the changeset is not relevant)
2. Open a pull request
3. Observe build failure

**Applicable Versions:**
- `webpack@4`
- `nodejs@17` and above

### PR DESCRIPTION
**Pull request by adunkman**
**Summary**

Upgrades Webpack to version 5 to fix #6502, which is preventing successful builds due to an interaction between Webpack 4 and Node versions 17 and above.

**Test plan**

Tests ran locally and had one failure with `yarn test:all`. Unfortunately, it appears tests are also not functioning in CI, as [Cypress is timing out waiting for the test server to boot](https://github.com/18F/netlify-cms/runs/7152815416) on 18F’s fork. 

<details>
<summary>1 test failure in <code>editorial_workflow_migration_spec_github_backend_rest.js</code></summary>
<br>

```
  Running:  editorial_workflow_migration_spec_github_backend_rest.js                       (1 of 43)
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating


  Github Backend Editorial Workflow Migration - REST API
Preparing environment for backend github
Running tests in "playback" mode - local data with be used
Setting up single test 'migrate from 2.9.7 to latest' for backend github
Seeding repository for backend github
Switching CMS to version '2.9.7'
[2022-07-05T14:21:05.333Z]  "GET /" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/103.0.5060.53 Safari/537.36"
[2022-07-05T14:21:10.504Z]  "GET /config.yml" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/103.0.5060.53 Safari/537.36"
    (Attempt 1 of 3) migrate from 2.9.7 to latest
Tearing down single test 'migrate from 2.9.7 to latest' for backend github
Setting up single test 'migrate from 2.9.7 to latest' for backend github
Seeding repository for backend github
Switching CMS to version '2.9.7'
[2022-07-05T14:21:19.099Z]  "GET /" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/103.0.5060.53 Safari/537.36"
[2022-07-05T14:21:19.344Z]  "GET /config.yml" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/103.0.5060.53 Safari/537.36"
[2022-07-05T14:21:19.379Z]  "GET /" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/103.0.5060.53 Safari/537.36"
[2022-07-05T14:21:19.441Z]  "GET /config.yml" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/103.0.5060.53 Safari/537.36"
    (Attempt 2 of 3) migrate from 2.9.7 to latest
Tearing down single test 'migrate from 2.9.7 to latest' for backend github
Setting up single test 'migrate from 2.9.7 to latest' for backend github
Seeding repository for backend github
Switching CMS to version '2.9.7'
[2022-07-05T14:21:27.968Z]  "GET /" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/103.0.5060.53 Safari/537.36"
[2022-07-05T14:21:28.073Z]  "GET /" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/103.0.5060.53 Safari/537.36"
[2022-07-05T14:21:28.139Z]  "GET /config.yml" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/103.0.5060.53 Safari/537.36"
    1) migrate from 2.9.7 to latest

  0 passing (31s)
  1 failing

  1) Github Backend Editorial Workflow Migration - REST API
       migrate from 2.9.7 to latest:
     AssertionError: Timed out retrying after 5000ms: Expected to find content: 'Entry saved' but never did.
      at Context.eval (http://localhost:8080/__cypress/tests?p=cypress/integration/editorial_workflow_migration_spec_github_backend_rest.js:765:8)



Tearing down single test 'migrate from 2.9.7 to latest' for backend github

  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        2                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      1                                                                                │
  │ Screenshots:  3                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     31 seconds                                                                       │
  │ Spec Ran:     editorial_workflow_migration_spec_github_backend_rest.js                         │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

https://user-images.githubusercontent.com/14930/177658056-da9f9ba5-13d4-435f-b6f4-2a2bbeca7f0a.mp4

What I’ve been able to gather: 
- At this stage of the test, v2.9.7 of netlify-cms [is loaded from unpkg.com](https://github.com/netlify/netlify-cms/blob/0c5efd19d6aa28e1e06a90c221ea0e8ac834ef33/cypress/utils/config.js#L27-L40).
- The test is failing during the setup phase, [when it is creating](https://github.com/netlify/netlify-cms/blob/0c5efd19d6aa28e1e06a90c221ea0e8ac834ef33/cypress/integration/common/editorial_workflow_migrations.js#L24) a [mocked entry](https://github.com/netlify/netlify-cms/blob/0c5efd19d6aa28e1e06a90c221ea0e8ac834ef33/cypress/integration/common/entries.js#L1-L7).
- Given that this pull request does not modify code from unpkg.com, does not modify mocked entries, and does not modify mocked responses, it does not seem likely that this pull request introduces this failure. 

I’d appreciate assistance if a contributor understands why this test may be failing. ❤️ 

----

</details>

**Checklist**

Please add a `x` inside each checkbox:

- [x] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [x] Code is formatted via running `yarn format`.
- [x] Tests are passing via running `yarn test`.
- [ ] The status checks are successful (continuous integration). Those can be seen below. (awaiting PR, but likely failing due to above mentioned issue with Cypress).

Fresh photo of Toulouse, for y’all! 🙂 More on #6498.

Photo of a Russian Blue and Domestic Shorthair cat mix, laying scrunched up on a windowsill in the sun. He is looking backwards over his left shoulder, wondering why he was interrupted from watching birds in the tree outside the window. He is wearing a rainbow collar.

### CODE REVIEW

File: packages/netlify-cms/webpack.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const path = require('path');
 require('dotenv').config({ path: path.join(__dirname, '..', '..', '.env') });
 const webpack = require('webpack');
-const FriendlyErrorsWebpackPlugin = require('friendly-errors-webpack-plugin');
+const FriendlyErrorsWebpackPlugin = require('@soda/friendly-errors-webpack-plugin');
```
Comment (adunkman) [inline packages/netlify-cms/webpack.config.js:4]: - `friendly-errors-webpack-plugin` does not support `webpack@5` and is [archived on GitHub](https://github.com/geowarin/friendly-errors-webpack-plugin).
- `@soda/friendly-errors-webpack-plugin` is a fork which supports `webpack@5`.

File: scripts/webpack.js [hunk 2]
```diff
@@ -44,8 +44,21 @@ function rules() {
...
+    ignoreEsprima: () =>
+      new webpack.IgnorePlugin({
+        resourceRegExp: /^esprima$/,
+        contextRegExp: /js-yaml/,
+      }),
+    ignoreMomentOptionalDeps: () =>
+      new webpack.IgnorePlugin({
+        resourceRegExp: /^\.\/locale$/,
+        contextRegExp: /moment$/,
+      }),
```
Comment (adunkman) [inline scripts/webpack.js:56]: The syntax for [IgnorePlugin changed](https://webpack.js.org/migrate/5/#clean-up-configuration) ([new documentation](https://webpack.js.org/plugins/ignore-plugin/#using-regular-expressions)): 

> If you are using `IgnorePlugin` with a regular expression as argument, it takes an `options` object now: `new IgnorePlugin({ resourceRegExp: /regExp/ })`.

File: scripts/webpack.js [hunk 3]
```diff
@@ -44,8 +44,21 @@ function rules() {
...
+    includePolyfills: () =>
+      new webpack.ProvidePlugin({
+        Buffer: ['buffer', 'Buffer'],
+        process: 'process',
+      }),
```
Comment (adunkman) [inline scripts/webpack.js:61]: Webpack 5 [no longer includes automatic polyfills for Node.js](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#automatic-nodejs-polyfills-removed): 

> In the early days, webpack's aim was to allow running most Node.js modules in the browser, but the module landscape changed and many module uses are now written mainly for frontend purposes. Webpack <= 4 ships with polyfills for many of the Node.js core modules, which are automatically applied once a module uses any of the core modules (i.e. the `crypto` module).
>
>While this makes using modules written for Node.js easier, it adds these huge polyfills to the bundle. In many cases these polyfills are unnecessary.
>
>Webpack 5 stops automatically polyfilling these core modules and focus on frontend-compatible modules. Our goal is to improve compatibility with the web platform, where Node.js core modules are not available.
>
>…
>
> It's possible to manually add a polyfill for a Node.js core module. An error message will give a hint on how to achieve that.

This (and the below `fallback` configuration lines) follow those hints.

**Comment by martinjagodic** [comment]
Dependencies are a mess right now, so they will be updated as part of a PR by maintainers.

**Pull request closed without merging**